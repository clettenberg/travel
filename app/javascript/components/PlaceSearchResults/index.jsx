import React from 'react'
import PropTypes from 'prop-types'

const props = {
  results: PropTypes.arrayOf(PropTypes.object).isRequired,
  selectedId: PropTypes.string,
  onPlaceSelection: PropTypes.func.isRequired
}

const defaultProps = {
  selectedId: null
}

const PlaceSearchResults = ({ results, selectedId, onPlaceSelection }) => (
  results.map((result) => {
    const inputId = `placeSearchResult-${result.place_id}`

    return (
      <div key={result.place_id} className='form-check'>
        <label className='form-check-label' htmlFor={inputId}>
          <input
            type='radio'
            className='form-check-input'
            name='placeSearchResults'
            id={inputId}
            value={result.place_id}
            checked={selectedId === result.place_id}
            onChange={onPlaceSelection}
          />
          <a
            href={`https://www.openstreetmap.org/${result.osm_type}/${result.osm_id}`}
            rel='noopener noreferrer'
            target='_blank'
          >
            {result.display_name}
          </a>
        </label>
      </div>
    )
  })
)

PlaceSearchResults.propTypes = props
PlaceSearchResults.defaultProps = defaultProps
export default PlaceSearchResults
