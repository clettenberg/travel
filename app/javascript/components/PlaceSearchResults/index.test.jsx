import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import PlaceSearchResults from './index'

describe('PlaceSearchResults', () => {
  it('should render correctly', () => {
    const results = [
      {
        place_id: 'foo', description: 'bar', osm_type: 'way', osm_id: '123456'
      },
      {
        place_id: 'another', description: 'fake-place', osm_type: 'way', osm_id: '123456'
      }
    ]

    const wrapper = shallow(
      <PlaceSearchResults
        results={results}
        selectedId='1234'
        onPlaceSelection={() => {}}
      />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
