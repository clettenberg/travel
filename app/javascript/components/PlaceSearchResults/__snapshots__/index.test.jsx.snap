// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceSearchResults should render correctly 1`] = `
Array [
  <div
    key="foo"
  >
    <label
      htmlFor="placeSearchResult-foo"
    >
      <input
        checked={false}
        id="placeSearchResult-foo"
        name="placeSearchResults"
        onChange={[Function]}
        type="radio"
        value="foo"
      />
      <a
        href="https://www.openstreetmap.org/way/123456"
        rel="noopener noreferrer"
        target="_blank"
      />
    </label>
  </div>,
  <div
    key="another"
  >
    <label
      htmlFor="placeSearchResult-another"
    >
      <input
        checked={false}
        id="placeSearchResult-another"
        name="placeSearchResults"
        onChange={[Function]}
        type="radio"
        value="another"
      />
      <a
        href="https://www.openstreetmap.org/way/123456"
        rel="noopener noreferrer"
        target="_blank"
      />
    </label>
  </div>,
]
`;
